File: from-2401-to-2500/2418-sort-the-people/solution.py
class Solution:
    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        return [
            name for name, height in 
            sorted([(name, height) for name, height in zip(names, heights)], key=lambda x: x[1], reverse=True)
               ]

File: from-2401-to-2500/2418-sort-the-people/solution.go
type info struct {
    name string
    height int
}

func sortPeople(names []string, heights []int) []string {
    n := len(names)
    people := make([]info, n)
    for i := 0; i < n; i ++ {
        people[i].name = names[i]
        people[i].height = heights[i]
    }
    sort.Slice(people, func(i, j int) bool {
        return people[i].height > people[j].height
    })
    result := make([]string, n)
    for i := 0; i < n; i ++ {
        result[i] = people[i].name
    }
    return result
}